File: src/main/java/org/geepawhill/contentment/grid/Vertical.java
package org.geepawhill.contentment.grid;

import org.geepawhill.contentment.geometry.PointPair;

public class Vertical
{
	
	private PointPair points;

	public Vertical(PointPair bounds, double percent)
	{
		double x = bounds.along(percent / 100d).getX();
		points = new PointPair( x, bounds.getFrom().getY(),x, bounds.getTo().getY());
	}
	
	public PointPair points()
	{
		return points;
	}
	
}

File: src/main/kotlin/org/geepawhill/contentment/grid/Vertical.kt
package org.geepawhill.contentment.grid

import org.geepawhill.contentment.geometry.PointPair

class Vertical(bounds: PointPair, percent: Double) {

    private val points: PointPair

    init {
        val x = bounds.along(percent / 100.0).x
        points = PointPair(x, bounds.from.y, x, bounds.to.y)
    }

    fun points(): PointPair {
        return points
    }

}
